ecommerce/api.py

```diff
@@ -244,3 +270,43 @@ def make_reference_id(order):
             A reference number for use with CyberSource to keep track of orders
     """
     return "{}{}-{}".format(_REFERENCE_NUMBER_PREFIX, settings.CYBERSOURCE_REFERENCE_PREFIX, order.id)
+
+
+def complete_order(order):
```

Comment:
I was thinking that this function would be called to perform all the side effects we want if an order was _successfully_ completed (maybe `complete_successful_order` or something like that would be a better name in that case). As it is now, this just shuffles some code from the view to this API method with its various responsibilities and I think it has limited usefulness. At a high level I think we could improve the view and API like this:

```python
# views.py
def post(self, request, *args, **kwargs):
  order = get_the_order(request)
  # receipt stuff...
  # bail out if errors are found or if the order doesn't need to change...
  if decision != CYBERSOURCE_DECISION_ACCEPT:
    handle_rejected_order(order)
  else:
    complete_successful_order(order)
  # this happens whether the order was rejected or accepted...
  sync_hubspot_deal_from_order(order)
  return Response()

# api.py
def complete_successful_order(order):
  # basically what you have already, plus setting the status flag, minus the API calls to hubspot and fluid/SMA...
```

That would mean setting the order status to "fulfilled" in this function, which I would still argue for despite it not matching what we have in xpro. One added benefit of rewriting the function in that way: we could call it from management commands or from our own scripts to accurately simulate a successful order completion.